paasta_tools/paasta_cli/cmds/local_run.py

```diff
@@ -574,40 +619,42 @@ def configure_and_run_docker_container(docker_client, docker_hash, service, args
     )
 
 
-def validate_environment():
-    """Validates whether the current directory is good for running
-    paasta local_run"""
-    if os.getcwd() == os.path.expanduser("~"):
-        sys.stderr.write(
-            'ERROR: Don\'t run this command from your home directory.\n'
-            'Try changing to the root of your working copy of the service.\n'
-        )
-        sys.exit(1)
-    if not os.path.isfile(os.path.join(os.getcwd(), 'Dockerfile')):
-        sys.stderr.write(
-            'ERROR: No Dockerfile in the current directory.\n'
-            'Are you in the root folder of the service directory? Does a Dockerfile exist?\n'
-        )
-        sys.exit(1)
+def local_makefile_present():
+    return True
 
 
 def paasta_local_run(args):
-    validate_environment()
+    # Try to autodetect if we should build the image locally or not
```

Comment:
This is my understanding of the current implementation:
- `local-run --pull` will always set `build=False`
- `local-run --build` will set `build=True` unless you also specify `--pull`, which has a higher priority
- `local-run` with neither argument will always set `build=True` and will print "Local Makefile detected..." if a Makefile is detected.

Having two flags makes more sense if the first clause reads more like this, otherwise `--build` is always the default unless `--pull` is specified.
```python
if args.build is None and args.pull is None:
    build = local_makefile_present()
```